TT3 compatibility

peterfabian opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This work builds on the great cleanup done by Rubik/Kirigami earlier this year. I did a tiny cleanup of colors from blocktheme.scss, where I moved the unnecessary definitions to TT2 theme's CSS to retain the same styling for TT2. 

Besides, did a quick analysis and comparison for all CSS rules and added the relevant ones from TT2 styling to TT3. Added also the WP class to buttons to enable all the style definitions that come with WP 6.1 and TT3.

Layout-wise, it's very similar to TT2. The colors got tricky here and there as some styles have effectively only 2 colors while some define 5 distinct colors, but I tried to make it work for all styles.

Style colors overview:
![TT3colors](https://user-images.githubusercontent.com/2207451/197823153-b07f8f82-a5f5-492f-a944-92241bd5f89b.jpg)


<!-- The next section is mandatory. If your PR doesn't require testing, please indicate that you are purposefully omitting instructions. -->

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Otherwise, please include detailed instructions on how these changes can be tested (including pre-conditions, configuration, steps to take and expected results). It may help to write your instructions using pseudocode -- as if you're telling a computer how to execute the test. -->

1. Install WP 6.1 RC and TT3
2. Install WC and activate this branch
3. Explore all the important screens: 

- Shop, 
- Single product for simple product
- Single product for variable products
- Examine gallery when having multiple images, photoswipe fullscreen image
- Check the Sale! badge looks reasonable in Shop and in single product page
- Check the tabs: Additional info/Details/Reviews
- Make sure the reviews are visible, rating looks reasonable
- Cart (try with or without shipping options set up), adding coupon
- Checkout
- Thank you page after submitting the order
- Log in to My account, with or without registration
- My account page
- Check out My Account submenus
- Test store notices (Added to cart on single product page, Error on the checkout page, Add coupon on the checkout page, Log in on the checkout page)
- Test store-wide "demo store" notice
- Check this in different viewport sizes
- Check all of this with different browsers
- Check all of this with each style in TT3 (Appearance > Editor > Click on Styles (half moon icon to the top right) and switch to another one)
![Screenshot 2022-10-25 at 17 49 45](https://user-images.githubusercontent.com/2207451/197823191-114bcbd7-dda0-4748-b7e6-20dc654f31b0.png)


<!-- End testing instructions -->

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

peterfabian committed:
Initial scaffolding

peterfabian committed:
Add wp element styling to buttons

peterfabian committed:
More buttons classes

peterfabian committed:
Removed colors with Woo-colored fallback

In themes like 2023 where foreground and background is undefined, this assignes incorrect colors.

peterfabian committed:
Buttons buttons my buttons!

peterfabian committed:
Blocktheme float layout fix

peterfabian committed:
TT3 initial CSS

peterfabian committed:
More styling.

peterfabian committed:
Moved definitions from blocktheme.scss

peterfabian committed:
Use WP function for button element instead of hardcoding.

peterfabian committed:
Not needed.

peterfabian committed:
Changelog.

peterfabian commented:
cc @Aljullu please let me know if you think these styles for blocks are needed for TT3 (code from the bottom of styling for TT2):
```
.wp-block-search {
	.wp-block-search__label {
		font-weight: normal;
	}
	.wp-block-search__input {
		padding: 0.9rem 1.1rem;
		border: 1px solid var(--wp--preset--color--black);
	}
	.wp-block-search__button {
		padding: 1rem 1.2rem;
	}
}

.wc-block-product-search {
	form {
		.wc-block-product-search__fields {
			display: flex;
			flex: auto;
			flex-wrap: nowrap;
			max-width: 100%;

			.wc-block-product-search__field {
				padding: 0.9rem 1.1rem;
				flex-grow: 1;
				border: 1px solid var(--wp--preset--color--black);
				font-size: inherit;
				font-family: inherit;
			}

			.wc-block-product-search__button {
				display: flex;
				background-color: var(--wp--preset--color--primary);
				color: #fff;
				border: 1px solid var(--wp--preset--color--black);
				padding: 1rem 1.2rem;
				margin: 0 0 0 0.7rem;
			}
		}
	}
}
```

peterfabian committed:
Escape the new class attribute.

peterfabian committed:
Fix the unit test.

peterfabian committed:
Incorrect indent wasn't intended.

peterfabian committed:
Fixed incorrect comment.

peterfabian committed:
PHPCS fixes.

Aljullu commented on plugins/woocommerce/includes/wc-template-functions.php:
```diff
@@ -2203,7 +2205,8 @@ function woocommerce_widget_shopping_cart_button_view_cart() {
 	 */
 	function woocommerce_widget_shopping_cart_proceed_to_checkout() {
-		echo '<a href="' . esc_url( wc_get_checkout_url() ) . '" class="button checkout wc-forward">' . esc_html__( 'Checkout', 'woocommerce' ) . '</a>';
+		$wp_button_class = wc_wp_theme_get_element_class_name( 'button' ) ? ' ' . wc_wp_theme_get_element_class_name( 'button' ) : '';
+		echo '<a href="' . esc_url( wc_get_checkout_url() ) . '" class="button checkout wc-forward' . esc_attr( $wp_button_class ) . '">' . esc_html__( 'Checkout', 'woocommerce' ) . '</a>';
```
I see this pattern repeated in several places, wondering if we could use `trim()` to simplify the code. Something along these lines:

```suggestion
		echo '<a href="' . esc_url( wc_get_checkout_url() ) . '" class="button checkout wc-forward' . esc_attr( trim( ' ' . wc_wp_theme_get_element_class_name( 'button' ) ) ) . '">' . esc_html__( 'Checkout', 'woocommerce' ) . '</a>';
```

Not sure if the code is much clearer, thought. :sweat_smile: So just leaving the suggestion here, but I don't have a strong preference anyway.

peterfabian replied:
Thanks! I was also slightly bothered by it, so was thinking about whether I should make a function out of it. The usage of `trim` is a nice trick.

Aljullu commented on plugins/woocommerce/templates/myaccount/orders.php:
```diff
@@ -67,7 +67,7 @@
 								if ( ! empty( $actions ) ) {
 									foreach ( $actions as $key => $action ) { // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
-										echo '<a href="' . esc_url( $action['url'] ) . '" class="woocommerce-button button ' . sanitize_html_class( $key ) . '">' . esc_html( $action['name'] ) . '</a>';
+										echo '<a href="' . esc_url( $action['url'] ) . '" class="woocommerce-button' . esc_attr( $wp_button_class ) . ' button ' . sanitize_html_class( $key ) . '">' . esc_html( $action['name'] ) . '</a>';
```
Just to confirm, since I'm not used to this codebase. We are inheriting `$wp_button_class` from a parent template? Because `$wp_button_class` is not declared in this file.

peterfabian replied:
Yup, it's defined over here: https://github.com/woocommerce/woocommerce/pull/35306/files#diff-d49488c34017d273db858b7b6eff58f22d95310991d26638e75793c6a1e98678R3213

The `echo` here is inside 2 foreach loops in the template, so didn't want to evaluate it each time, and we also want to make templates as dumb as possible, so having conditions and other code outside of them unless necessary. Which makes me think I should also remove the other conditionals from templates 🤔

Aljullu commented on plugins/woocommerce/includes/theme-support/class-wc-twenty-twenty-three.php:
```diff
@@ -0,0 +1,82 @@
+/**
+ * Twenty Twenty-Three support.
+ *
+ * @since   7.0.2
```
In many places we specify `@version 7.1.0`, I guess this will need to be unified once we know whether that's included in a patch release?

peterfabian replied:
Yup, I'll search and replace it once we figure it out. Good point.

peterfabian replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/35306/commits/f543cc05658110ea8d08e3d261d536d0c24d69d5

Aljullu commented on plugins/woocommerce/client/legacy/css/twenty-twenty-two.scss:
```diff
@@ -220,6 +246,24 @@ $tt2-gray: #f7f7f7;
+		&:disabled[disabled]:hover {
+			background-color: var(--wp--preset--color--foreground, $primary);
+			color: var(--wp--preset--color--background, $primarytext);
+			opacity: 0.5;
```
Nit: `opacity: 0.5` seems to be unnecessarily repeated in both files:

https://github.com/woocommerce/woocommerce/pull/35306/files#diff-4d8249aa26cad0c3201e91cebddb2c9adcf41c37c5de6f8f723b4936d2619892L56

Same for other opacities added to this file. Was it intentional?

peterfabian replied:
I was just being a bit lazy 😬. Removed it from TT2, since I wanted to only remove the colors from blocktheme.css. Fixed in https://github.com/woocommerce/woocommerce/pull/35306/commits/4b23164b0321515260e0ceb36af6ed1d70220430

Aljullu reviewed (commented):
Thanks for working on this, @peterfabian!

I did an initial review. I didn't finish it yet, but leaving some early feedback for now.

> cc @Aljullu please let me know if you think these styles for blocks are needed for TT3 (code from [the bottom of styling for TT2](https://github.com/woocommerce/woocommerce/blob/7.0.0/plugins/woocommerce/client/legacy/css/twenty-twenty-two.scss#L1108-L1147)):

It should be possible to remove everything under `.wc-block-product-search` because in WP 6.1 the Product Search block will be replaced with a GB Search block variation (https://github.com/woocommerce/woocommerce-blocks/pull/6191), so we already inherit all styling from the GB block.

Regarding the `.wp-block-search` rules, they impact the GB Search block (and the new Product Search variation). I would personally remove them, because I don't think it's WC's job to style that block.

peterfabian committed:
Avoid duplication of CSS rules.

Aljullu commented on plugins/woocommerce/client/legacy/css/twenty-twenty-three.scss:
```diff
@@ -0,0 +1,1063 @@
+			//&:hover,
+			//&:visited {
+			//	color: var(--wp--preset--color--black);
+			//}
```
Should commented code be removed?

peterfabian replied:
yup, thanks for noticing

Aljullu reviewed (commented):
Hey @peterfabian, I did another round of testing, so leaving here everything I could find. I will leave it up to Proton to decide whether those issues are blocking or not.

---

That's a personal opinion, but I would remove the background-color from `.woocommerce mark`, it looks kinda odd:

![imatge](https://user-images.githubusercontent.com/3616980/198234788-656c303b-9bd3-4ba3-a480-a9ca6c2a92c3.png)

![imatge](https://user-images.githubusercontent.com/3616980/198236676-a7310b45-7356-4bd5-988c-b948fd7316e7.png)

It looks good in some color combinations (see #31631), but not in many others and I would tend to the safe side. If we want to highlight those values, we can use `font-weight` or an alternative approach.

---

Tabs in the product template are not legible when active:

![imatge](https://user-images.githubusercontent.com/3616980/198237131-532a5244-ea98-43d6-89fa-71dda39dcbb6.png)

I think this should be fixed at the `woocommerce.css` level: `.woocommerce div.product .woocommerce-tabs ul.tabs li.active` sets a `background-color` but not a `color`. That's usually dangerous because `color` is inherited and there is no guarantee there will be contrast with `background-color`. IMO every time there is a `background-color` declaration, there should be a `color` one as well.

---

If you go to `/wp-admin/customize.php` > WooCommerce > Product Catalog, set Shop page display to Show categories and then go to the Shop page in the frontend, you will notice the categories names are huge:

![imatge](https://user-images.githubusercontent.com/3616980/198243313-9dcce9a9-165e-46b3-87de-d7f7ffb9b4dc.png)

---

The _Have a coupon?_ notice in the Checkout is not legible with some style variations:

![imatge](https://user-images.githubusercontent.com/3616980/198243849-79e71529-de78-401e-bd73-698addf1a4af.png)

peterfabian committed:
More CSS adjustments.

peterfabian commented:
Thanks for testing, @Aljullu !

> I would remove the background-color from .woocommerce mark

I totally agree, never liked the background colors there tbh. Switched to bold text instead.

> Tabs in the product template are not legible when active

I was specifically fixing these colors but it seems it's not the correct version that made the cut into the PR. Should be fixed now.

> I think this should be fixed at the woocommerce.css level: .woocommerce div.product .woocommerce-tabs ul.tabs li.active sets a background-color but not a color...

That's probably a good general rule, added. 👍 

> The Have a coupon? notice in the Checkout is not legible with some style variations:
> Tabs in the product template are not legible when active

Honestly, I'm not quite sure what to do here. When you look at the style colors stacked in the PR description image above, there is no 2-color combination that 
a) is different from base-contrast (so can be used as a combo with different background than main background)
b) has good legibility in all styles. 

Even if it works in most, then I usually have problems with e.g. Electric style which actually defines just 2 colors. Additionally, I can't use any SCSS color functions like `lighten`, because those don't work with `var(--wp--color--preset--xyz)`.  

I guess the only option is to use something else than background color to style these, perhaps a border. I also examined using  `mix-blend-mode` and `background-blend-mode`, but it seems the elements inside like buttons then look weird, so opted for a semi-transparent gray. It doesn't look awesome but let's say it looks _acceptable_. If you have any ideas, please let me know!

> categories names are huge

Thanks for noticing, I usually don't use these, so I missed it. Should be fixed now.

All fixes should be included in [`8401529` (#35306)](https://github.com/woocommerce/woocommerce/pull/35306/commits/84015296c64e8c220f55cb451e4d99fe12f8d47b)

peterfabian committed:
Fix version tags

peterfabian commented:
Note: the currently broken e2e test fails in Admin area, so should be unaffected by these changes.

Konamiman commented:
Possible issues in the shop page:

- Light green on white for product names not very readable
- Product names and buttons misaligned depending on the presence or absence of ratings and prices (and on the length of the product name itself)
- "Sale" badge quite far away when there is no product image

![image](https://user-images.githubusercontent.com/937723/198532900-b162cf7a-386c-46ea-ac96-712ff42dedef.png)

![image](https://user-images.githubusercontent.com/937723/198532973-9200c7e5-fcb7-48af-8369-b46af29a77d4.png)

Konamiman commented:
In product page:

- Misaligned variation selection and quantity dropdowns
- Weird black line between "Additional information" and "Reviews" tabs
- Centered feature names, maybe they should be left aligned instead
- Misaligned buttons for "Related products"

![image](https://user-images.githubusercontent.com/937723/198534035-cd0ee0a7-584b-4878-a4c8-4d515d693b70.png)

![image](https://user-images.githubusercontent.com/937723/198534577-18c5cb0e-77d0-42e6-9d78-b4803744ace5.png)

Konamiman commented:
Not sure if this is an issue, but product gallery images looks a bit like a puzzle:

![image](https://user-images.githubusercontent.com/937723/198536251-bc466a0f-c7b7-4cac-afa2-fd051e792db3.png)

Konamiman commented:
Reviews don't look nice when there's a long text. Also maybe there should be some more space between reviews.
![image](https://user-images.githubusercontent.com/937723/198537014-421e296e-3cf0-4d50-8562-6e50e8cb6bf7.png)

Konamiman commented:
Adding to cart: notice looks good, the rest has same problems as single product page:

![image](https://user-images.githubusercontent.com/937723/198538344-f57418e0-3812-4be2-8b79-8ec5d3a95984.png)

Konamiman commented:
Cart: looking good in general, but I wonder if these black lines between items should be there (or maybe they should look dimmer):

![image](https://user-images.githubusercontent.com/937723/198539007-51347f12-900e-4221-a147-a4f5692c71fc.png)

Konamiman commented:
Maybe this is nitpicking but: maybe too much space above and below "Billing details"/"Your order" titles on checkout?

![image](https://user-images.githubusercontent.com/937723/198541068-c57aac10-5ccd-4734-a9c4-923aef4bb469.png)

Konamiman commented:
PayPal logo image looks a bit crammed in the payment options section of the checkout page:
![image](https://user-images.githubusercontent.com/937723/198541277-c953a919-2ac5-4b3d-98c0-05f3c4336d6a.png)

peterfabian committed:
Shop: product title visibility

peterfabian committed:
Shop: button alignment.

Konamiman commented:
My account - Orders (and similarly for downloads):

- Maybe too much of blank space between left menu and list items
- Table header has no style and looks like a regular data row (maybe this is on purpose?)
- As in the case of the cart page, maybe lines between items could disappear or look dimmer

![image](https://user-images.githubusercontent.com/937723/198545727-d0a6017e-a0ee-45eb-8ba4-5a9394e37531.png)

Konamiman commented:
My account - Orders with very narrow viewport: no space at all before the "View" buttons.

![image](https://user-images.githubusercontent.com/937723/198545958-5483c971-49bf-44f9-a132-79f4c257ec2a.png)

peterfabian commented:
> shop page:

> Light green on white for product names not very readable
Fixed in [196b26c](https://github.com/woocommerce/woocommerce/pull/35306/commits/196b26cd9eb6912b773f6f3868b21772e6286c08).

> Product names and buttons misaligned depending on the presence or absence of ratings and prices (and on the length of the product name itself)

It's the same in TT2, but fixed in [0cfb3c1](https://github.com/woocommerce/woocommerce/pull/35306/commits/0cfb3c16549ce323d4fa3b3007dcc8efe33d173b)

> "Sale" badge quite far away when there is no product image

Yeah, it's at the edge of the image, hard to style it if the page background is the same as the image background. Seems acceptable to me.

peterfabian committed:
Improved Shop layout on narrow screens.

Konamiman commented:
Login/registration page with narrow vieport: no margin left and right of the "Register" form fields, but there is some space for the "Login" fields (the general rule for this theme seems to be no margin at all for narrow viewports):

![image](https://user-images.githubusercontent.com/937723/198546593-24ddbecf-4fe4-4706-a356-350785532377.png)

peterfabian committed:
Removed weird shadows and borders

peterfabian committed:
Make product attributes stick to the left.

peterfabian committed:
Correct dash font size for var product price

Konamiman commented:
When a product is out of stock the "Read more" button that replaces "Add to cart" appears misaligned:

![image](https://user-images.githubusercontent.com/937723/198565016-5d87fe39-f057-49c2-8268-1a65a2cf5323.png)

peterfabian committed:
Styling of variation chooser

peterfabian commented:
>  variable product page:

> Misaligned variation selection and quantity dropdowns

Should be fixed in [`80c3e30` (#35306)](https://github.com/woocommerce/woocommerce/pull/35306/commits/80c3e305ed5e47c402639c194105b72ab194c9eb)

> Weird black line between "Additional information" and "Reviews" tabs

Removed

> Centered feature names, maybe they should be left aligned instead

Strange, I actually see them aligned to the left:
![Screenshot 2022-10-28 at 11 42 40](https://user-images.githubusercontent.com/2207451/198557433-ae317564-8333-439f-b771-b7e7cba34a3f.png)
But set the `text-align` to make sure it stays on the left, the default behaviour is probably centering it. 

> Misaligned buttons for "Related products"

Fixed with the previous commit.

Konamiman commented:
>>Centered feature names, maybe they should be left aligned instead

>Strange, I actually see them aligned to the left

Look closely, they aren't 🙂

![image](https://user-images.githubusercontent.com/937723/198566273-bd156121-74b5-471f-8d8a-1abd5345f1b5.png)

peterfabian committed:
Style all buttons in the shop the same.

peterfabian commented:
> When a product is out of stock the "Read more" button that replaces "Add to cart" appears misaligned:

Fixed in [`d4abbf3` (#35306)](https://github.com/woocommerce/woocommerce/pull/35306/commits/d4abbf3c692833646b6a3dc3bcef71f09505d64c)

peterfabian commented:
>>>Centered feature names, maybe they should be left aligned instead

>>Strange, I actually see them aligned to the left

>Look closely, they aren't 🙂

Good eye! Should be fixed now :)

peterfabian committed:
Better formatting for reviews

peterfabian commented:
> Reviews don't look nice when there's a long text. Also maybe there should be some more space between reviews.

Fixed in [`92d97e4` (#35306)](https://github.com/woocommerce/woocommerce/pull/35306/commits/92d97e40722ede0800e4fcbcdfc05f79d81b83a1)

peterfabian committed:
Improve cart styling: border between lines using theme color.

peterfabian commented:
> Cart: looking good in general, but I wonder if these black lines between items should be there (or maybe they should look dimmer):

Changed the color to contrast variable, which hopefully should better complement each theme. [`911a375` (#35306)](https://github.com/woocommerce/woocommerce/pull/35306/commits/911a375c07f5ce5a55c4e3b03a4b1a51f1d8bfab)

Konamiman commented:
Aubergine style: black to purple transition sometimes in the middle of a sentence, not sure if on purpose. Also the transition isn't happening at the same place across pages.

![image](https://user-images.githubusercontent.com/937723/198573245-2264f894-c439-4506-b25b-7f31d01b8197.png)

![image](https://user-images.githubusercontent.com/937723/198573791-b41437b3-c03a-4c2f-a42a-8105ec63a6b8.png)

Konamiman commented:
Canary style: text in some wbuttons wraps oddly.

![image](https://user-images.githubusercontent.com/937723/198574191-f1d1cf0d-adea-4961-a982-8dc67debe40c.png)

![image](https://user-images.githubusercontent.com/937723/198574236-4d5c6ce7-76ba-42b0-9861-9873af1c89ac.png)

Konamiman commented:
Pitch style: links look a bit too dark in general, links in notices are hardly readable.

![image](https://user-images.githubusercontent.com/937723/198574745-3ab29fae-5c2b-4928-9e87-c158314b265b.png)

![image](https://user-images.githubusercontent.com/937723/198574814-d1461247-7166-4495-a4bd-da202a6d80d8.png)

Konamiman commented:
Sherbet style: buttons are hardly recognizable as such (they are missing the black border they have in other pages?), and same problem with "Cancel" button as in the Canary style.

![image](https://user-images.githubusercontent.com/937723/198575043-392c340b-345a-4ad1-b19d-d818bf4c9c9c.png)

Konamiman commented:
Whisper style: in My Account the current section has a weird extra underline, and in Orders the "Cancel" button has also a text wrapping issue.

![image](https://user-images.githubusercontent.com/937723/198575655-22bf68d2-1ff2-4796-a495-9dbda8a0c5e3.png)

![image](https://user-images.githubusercontent.com/937723/198575690-fa8e3f17-9ecd-4aa5-b1d7-05ee8ae3f05a.png)

peterfabian committed:
My Account tables styling

peterfabian commented:
> My account - Orders (and similarly for downloads):

> Maybe too much of blank space between left menu and list items

Leaving it as is for now.

> Table header has no style and looks like a regular data row (maybe this is on purpose?)

Updated in [`57a9797` (#35306)](https://github.com/woocommerce/woocommerce/pull/35306/commits/57a979719305c7baea6954ab06ce97778dcec159)

> As in the case of the cart page, maybe lines between items could disappear or look dimmer

Switched them to the color of contrast for the theme.

peterfabian committed:
MyAccount > Orders View narrow view action button fix

peterfabian commented:
> My account - Orders with very narrow viewport: no space at all before the "View" buttons.

Fixed in [`55a9a0d` (#35306)](https://github.com/woocommerce/woocommerce/pull/35306/commits/55a9a0db81b7ffb0659ee95763885a74c645ba83)

peterfabian commented:
> Login/registration page with narrow vieport: no margin left and right of the "Register" form fields, but there is some space for the "Login" fields (the general rule for this theme seems to be no margin at all for narrow viewports)

For me, they're both part of `woocommerce` div that is inside another div with class `.has-global-padding` that gives it the padding:
![Screenshot 2022-10-28 at 13 44 32](https://user-images.githubusercontent.com/2207451/198579765-44698c80-79fd-46cc-b00f-f6d9e88f4c99.png)

peterfabian commented:
> Aubergine style: black to purple transition sometimes in the middle of a sentence, not sure if on purpose. Also the transition isn't happening at the same place across pages.

I think this depends on the length of the page content, e.g. this is what I'm getting for WP pages with no WC shortcodes 
![Screenshot 2022-10-28 at 13 50 35](https://user-images.githubusercontent.com/2207451/198580740-defc7c65-dc94-4453-9f27-2f814e2651c3.png)

peterfabian committed:
Make buttons less wide to account for narrow theme styles.

peterfabian commented:
> Canary style: text in some wbuttons wraps oddly.

[`b03631c` (#35306)](https://github.com/woocommerce/woocommerce/pull/35306/commits/b03631c1bda203ea64dec2988c3cb031fe4c774c)

Reduced the padding of buttons on the Shop page so that narrow content styles would have nicer wrapping:

![Screenshot 2022-10-28 at 14 02 44](https://user-images.githubusercontent.com/2207451/198582671-291610a8-e8aa-43d7-a39f-157aff8febd5.png)

peterfabian committed:
Large floated buttons are sometimes larger than the notice.

peterfabian committed:
Better color for links in notices and fix coupon buttons in cart.

peterfabian commented:
> Pitch style: links look a bit too dark in general, links in notices are hardly readable.

Updated in [`62308a1` (#35306)](https://github.com/woocommerce/woocommerce/pull/35306/commits/62308a1e7f572e0d2d99a06cb084bee150211f23) but it's tricky with the colors, I hope it will look reasonable in all styles.

peterfabian committed:
Reduce padding for My Account > order table

But this only goes so far. If e.g. the translation is longer, it will look weird anyway.

peterfabian commented:
> Sherbet style: in My Account - Orders buttons are hardly recognizable as such (they are missing the black border they have in other pages?), and same problem with "Cancel" button as in the Canary style.

[`4cefefb` (#35306)](https://github.com/woocommerce/woocommerce/pull/35306/commits/4cefefbc56c2ac38efbd9680dae558b3b70ca56e)
Fixed the border and reduced padding: 

![Screenshot 2022-10-28 at 14 38 42](https://user-images.githubusercontent.com/2207451/198588817-ab4b9e29-3a4f-435f-9eed-8e6684d8c414.png)

peterfabian committed:
More fixes!

peterfabian commented:
> Whisper style: in My Account the current section has a weird extra underline, and in Orders the "Cancel" button has also a text wrapping issue.

This should be fixed now as well.

peterfabian committed:
Some styles are not working with this color combo.

peterfabian committed:
Align attribute names to the left

peterfabian commented:
> Similarly to the "Additional information" tab, the attribute names in the variation selectors should probably be left aligned too

Fixed in [2742331](https://github.com/woocommerce/woocommerce/pull/35306/commits/27423318d9a7c39cd008ffe9121c52c5f85316ff)

peterfabian committed:
base color doesn't always work

helgatheviking commented on plugins/woocommerce/templates/single-product/add-to-cart/simple.php:
```diff
@@ -46,7 +46,7 @@
 		?>
 
-		<button type="submit" name="add-to-cart" value="<?php echo esc_attr( $product->get_id() ); ?>" class="single_add_to_cart_button button alt"><?php echo esc_html( $product->single_add_to_cart_text() ); ?></button>
+		<button type="submit" name="add-to-cart" value="<?php echo esc_attr( $product->get_id() ); ?>" class="single_add_to_cart_button button alt<?php echo esc_attr( wc_wp_theme_get_element_class_name( 'button' ) ? ' ' . wc_wp_theme_get_element_class_name( 'button' ) : '' ); ?>"><?php echo esc_html( $product->single_add_to_cart_text() ); ?></button>
```
@peterfabian is there a need to call `wc_wp_theme_get_element_class_name( 'button' )` twice? Could this be

```
<?php echo esc_attr( wc_wp_theme_get_element_class_name( 'button' ) ); ?>
```

rrennick replied:
@helgatheviking It would need a leading space but that could be added after `alt`.

helgatheviking replied:
Gotcha. Seems like worse case is there's a trailing space if you remove the duplication...  `class="single_add_to_cart_button button alt ">`
